pkg/generate/app/app.go

```diff
@@ -216,6 +219,9 @@ func (r *ImageRef) RepoName() string {
 
 // SuggestName suggests a name for an image reference
 func (r *ImageRef) SuggestName() (string, bool) {
+	if r != nil && len(r.ObjectName) > 0 {
```

Comment:
Leave the `if r == nil || ...` check on top so that you will just need to `if len(r.ObjectName)` for this...